nano/node/lmdb.cpp

```diff
@@ -1205,10 +1205,10 @@ void nano::mdb_store::upgrade_v11_to_v12 (nano::transaction const & transaction_
 	mdb_drop (env.tx (transaction_a), checksum, 1);
 }
 
-void nano::mdb_store::upgrade_v12_to_v13 ()
+void nano::mdb_store::upgrade_v12_to_v13 (size_t const batch_size)
 {
 	size_t cost (0);
-	size_t const max (16384);
+	size_t const max (batch_size);
```

Comment:
You could pass max here `void nano::mdb_store::upgrade_v12_to_v13 (size_t const max)` or make `if (cost >= batch_size)`